packages/components/Menu/src/Menu/useMenu.android.ts

```diff
@@ -0,0 +1,250 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import type { LayoutChangeEvent, View } from 'react-native';
+import { Animated, Dimensions, Easing, I18nManager, StatusBar } from 'react-native';
+
+import type { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+import type { MenuProps, MenuState } from './Menu.types';
+import { AndroidMenuStates } from './Menu.types';
+import { useMenuContext } from '../context/menuContext';
+// Due to how events get fired we get double notifications
+// for the same event causing us to immediately reopen
+// a menu when we close it. Adding in a delay to prevent
+// this behavior.
+const delayOpen = 150;
+let lastCloseTimestamp = -1;
+
+const EASING = Easing.bezier(0.4, 0, 0.2, 1);
+const SCREEN_INDENT = 16;
+
+export const useMenu = (props: MenuProps): MenuState => {
+  const triggerRef = React.useRef();
+  const context = useMenuContext();
+  const isSubmenu = context.triggerRef !== null;
+  const isOpenControlled = typeof props.open !== 'undefined';
+  const [checked, onCheckedChange] = useMenuCheckedState(props);
+
+  const _container = useRef<View>(null);
+  const [menuState, setMenuState] = React.useState<AndroidMenuStates>(AndroidMenuStates.Hidden);
+  const [maxMenuHeight] = useState(250);
+  const [anchorWidth, setAnchorWidth] = React.useState<number>(0);
```

Comment:
This code is non verbose and hard to review - please consider:
- Splitting the code into smaller chunks/ groups which are locally related.
- Code comments across what is the calculation intending to do

Reply:
Added comments to each of the Function and groups what there functions are. 
As this code is only for Android platform specific and don't have reusable hooks keeping here everything with comments.